### PR TITLE
Add smely socks

### PR DESCRIPTION
**Pull request by MathyFurret**
https://www.smogon.com/forums/threads/mimikyu.3715930/

### CODE REVIEW

File: data/mods/ssb/abilities.ts [hunk 1]
```diff
@@ -518,6 +518,58 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
...
+			if (!target.runImmunity(move.type)) return;
+			return 0;
+		},
+		onUpdate(pokemon) {
+			if (['mimikyu', 'mimikyutotem'].includes(pokemon.species.id) && this.effectState.busted) {
+				const speciesid = pokemon.species.id === 'mimikyutotem' ? 'Mimikyu-Busted-Totem' : 'Mimikyu-Busted';
+				pokemon.formeChange(speciesid, this.effect, true);
+				this.damage(pokemon.baseMaxhp / 8, pokemon, pokemon, this.dex.species.get(speciesid));
+				this.boost({atk: 1, spe: 1});
+				const img = "https://www.smogon.com/forums/attachments/alcremie-png.494778/";
```
Comment (HisuianZoroark) [inline data/mods/ssb/abilities.ts:565]: not everyone is going to see that

File: data/mods/ssb/conditions.ts [hunk 2]
```diff
@@ -447,6 +447,19 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
 			}
 		},
 	},
+	smelysocks: {
+		noCopy: true,
+		onStart() {
+			const img = "https://www.smogon.com/forums/attachments/meow_gm-png.494777/";
```
Comment (HisuianZoroark) [inline data/mods/ssb/conditions.ts:453]: needs a better link

File: data/mods/ssb/moves.ts [hunk 3]
```diff
@@ -968,6 +968,34 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Curse', target);
+			this.add('-anim', source, 'Block', target);
+		},
+		onHit(target, source, move) {
+			if (!target.addVolatile('curse')) return false;
+			this.directDamage(source.maxhp / 2, source, source);
+			target.addVolatile('trapped', source, move, 'trapper');
+		},
```
Comment (HisuianZoroark) [inline data/mods/ssb/moves.ts:992]:
```suggestion
		onHit(target, source, move) {
			let success = false;
			if (!target.volatiles['curse']) {
				this.directDamage(source.maxhp / 2, source, source);
				target.addVolatile('curse');
				success = true;
			}
			return target.addVolatile('trapped', source, move, 'trapper') || success;
		},
```

File: data/mods/ssb/conditions.ts (hunk 2 above)
Comment (HisuianZoroark) [inline data/mods/ssb/conditions.ts:453]:
```suggestion
			const img = "https://media.discordapp.net/attachments/764667730468536320/1079168557553831947/alcremie.png";
```

File: data/mods/ssb/abilities.ts (hunk 1 above)
Comment (HisuianZoroark) [reply data/mods/ssb/abilities.ts:565]:
```suggestion
				const img = "https://media.discordapp.net/attachments/764667730468536320/1079168557553831947/alcremie.png";
```

File: data/mods/ssb/moves.ts (hunk 3 above)
Comment (HisuianZoroark) [reply data/mods/ssb/moves.ts:992]: better order of events + doesn't auto fail if the mon can still trap but not inflict curse

**Comment by MathyFurret** [comment]
@HisuianZoroark Are the move flags OK? I copied them from Curse but since this one doesn't have any target weirdness it might be able to have more normal interactions.

**Review by HisuianZoroark — APPROVED** [review]
No complaints here

**Pull request merged**